### PR TITLE
Customisable control buttons

### PR DESCRIPTION
**Pull request by callumjt**
Customisable control buttons

allows you to customise the play, pause (if you have it enabled) and the stop buttons

### CODE REVIEW

**Comment by DNin01** [comment]
Nice work, but most people aren't really looking at customizing these small details. Try aiming for areas of the website that people often want to be able to customize, like the colors of the whole website (take the `Website dark mode and customizable colors` addon, for example).

Also, if you find something worth customizing but you don't feel it's big enough for its own addon, you can put the option(s) in another related addon like `Website dark mode and customizable colors` or `Editor dark mode and customizable colors`.

**Comment by callumjt** [comment]
Oh ok :(

**Comment by Samq64** [comment]
You could make an issue for your suggestions and maybe after some discussion it might be added as a new addon or another existing addon if it's decided it's a feature fit form Scratch Addons.

**Comment by callumjt** [comment]
Ye I'll probably do that in the future

**Comment by DNin01** [comment]
You might also find the [contributing guidelines](https://github.com/ScratchAddons/ScratchAddons/blob/master/.github/CONTRIBUTING.md) useful.

Since your `master` branch still has changes on it, you may want to reset it by either deleting your fork, creating a new branch with `ScratchAddons/ScratchAddons:master` selected as the branch source and make it your new default, or some other way so it's clean for the next time you contribute.

**Comment by callumjt** [comment]
Ah ok. Sorry about that and thank you

**Comment by Secret-chest** [comment]
Please reopen, I think they are seen very often so they would be worth it

**Comment by Samq64** [comment]
Re-opeing this doesn't necessarily mean it will be merged.

**Comment by callumjt** [comment]
should i redo the pull request when i update the fork to be correct?

if i redo it there will be more features

**Comment by DNin01** [comment]
> Please reopen, I think they are seen very often so they would be worth it

I feel that this much customization will be overwhelming for the average user, but if we decide to do this, then we probably don't need options to change colors (since few people will change that and that area is kind of already covered by Editor dark mode and customizable colors) and we could have fewer, more desirable customization options. Perhaps a setting for a small version of the green flag, pause button, and stop sign.

Since Samq64 reopened this, let's just see how this goes.

**Comment by DNin01** [comment]
> should i redo the pull request when i update the fork to be correct?
> 
> 
> 
> if i redo it there will be more features

Sure, you can make a new pull request if you would like to start a new branch. Or if you want to modify your current code by updating your `master` branch, this pull request will be updated along with it, so you can keep it.

**Comment by callumjt** [comment]
> > should i redo the pull request when i update the fork to be correct?
> > if i redo it there will be more features
> 
> Sure, you can make a new pull request if you would like to start a new branch. Or if you want to modify your current code by updating your `master` branch, this pull request will be updated along with it, so you can keep it.

ah ok thank you

**Comment by callumjt** [comment]
> > > Please reopen, I think they are seen very often so they would be worth it
> > 
> > I feel that this much customization will be overwhelming for the average user, but if we decide to do this, then we probably don't need every setting we have now to stay (since few people will change all of them and background colors are kind of already covered by Editor dark mode and customizable colors) and we could have fewer, more desirable customization options. Perhaps a setting for a small version of the green flag, pause button, and stop sign.
> > 
> > Since Samq64 reopened this, let's just see how this goes.
> 
> Is 2 settings too much? You don't have to change them!

I have added a few more things and I'm planning to add one more thing though I'm not sure how I'll do it

**Comment by Secret-chest** [comment]
What about changing their colours?

**Comment by callumjt** [comment]
just made a new update :D 

new features:
-option to change border radius
-option to disable the hover effect
-made the colour options only show if you have the setting enabled. hopefully this is less overwhelming when you see the addon

**Comment by callumjt** [comment]
I think imma just stop adding things cause tbh I don't think it's gonna get accepted :/

**Comment by WorldLanguages** [comment]
We don't usually offer addon settings that allow users to change 1 CSS property at a time. For users that want that level of customizability we often recommend writing Stylus userstyles.

File: addons/customisable-control-buttons/addon.json [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        "max": 100
+      },
+      {
+        "name": "Remove hover (must refresh to apply)",
+        "id": "hover",
+        "type": "boolean",
+        "default": false
+      },
+      {
+        "name": "Buttons background (must refresh to apply)",
```
Comment (WorldLanguages) [inline addons/customisable-control-buttons/addon.json:74]: The "must refresh to apply" can probable be fixed with `"updateUserstylesOnSettingsChange": true` in the root of the manifest

**Comment by WorldLanguages** [comment]
@Secret-chest Are you interested on changing the color of the icons? Maybe that would be feasible. But we surely don't want to provide a setting to customize how many px separate each button from the next.

File: addons/customisable-control-buttons/addon.json (hunk 1 above)
Comment (callumjt) [reply addons/customisable-control-buttons/addon.json:74]: ah ok thank you

**Comment by callumjt** [comment]
> 

should i remove it?

**Comment by callumjt** [comment]
> > @Secret-chest Are you interested on changing the color of the icons? Maybe that would be feasible. But we surely don't want to provide a setting to customize how many px separate each button from the next.
> 
> should i remove it?

also im not really sure how to change the icon without changing the entire background-color

**Comment by Secret-chest** [comment]
> > > @Secret-chest Are you interested on changing the color of the icons? Maybe that would be feasible. But we surely don't want to provide a setting to customize how many px separate each button from the next.
> > 
> > should i remove it?
> 
> also im not really sure how to change the icon without changing the entire background-color

Are they SVG? If then you can change the `fill` property

**Comment by callumjt** [comment]
> > > > @Secret-chest Are you interested on changing the color of the icons? Maybe that would be feasible. But we surely don't want to provide a setting to customize how many px separate each button from the next.
> > > 
> > > should i remove it?
> > 
> > also im not really sure how to change the icon without changing the entire background-color
> 
> Are they SVG? If then you can change the `fill` property
Ah ok thank you

**Comment by callumjt** [comment]
> > > > > @Secret-chest Are you interested on changing the color of the icons? Maybe that would be feasible. But we surely don't want to provide a setting to customize how many px separate each button from the next.
> > > > 
> > > > should i remove it?
> > > 
> > > also im not really sure how to change the icon without changing the entire background-color
> > 
> > Are they SVG? If then you can change the `fill` property
> 
> Ah ok thank you 
> 

Wait but how would I go about this? Do i let them change every icons color individually?

**Comment by Secret-chest** [comment]
> > > > > > @Secret-chest Are you interested on changing the color of the icons? Maybe that would be feasible. But we surely don't want to provide a setting to customize how many px separate each button from the next.
> > > > > 
> > > > > should i remove it?
> > > > 
> > > > also im not really sure how to change the icon without changing the entire background-color
> > > 
> > > Are they SVG? If then you can change the `fill` property
> > 
> > Ah ok thank you 
> > 
> 
> Wait but how would I go about this? Do i let them change every icons color individually?

Of course, they are differently coloured in vanilla so you need to provide separate options for flag, pause and stop

**Comment by Secret-chest** [comment]
What about merging it with reverse order of project controls?

**Comment by mxmou** [comment]
> Are they SVG? If then you can change the `fill` property

That won't work because they're `img` elements.

**Comment by Secret-chest** [comment]
> > Are they SVG? If then you can change the `fill` property
> 
> That won't work because they're `img` elements.

Oh, so should we reimplement them using inline SVG?

**Comment by callumjt** [comment]
what if i were to merge this with "Reverse order of project controls" and change the name?

**Comment by callumjt** [comment]
> What about merging it with reverse order of project controls?

oh i didnt see this

**Comment by WorldLanguages** [comment]
> what if i were to merge this with "Reverse order of project controls" and change the name?

Merge "this"? What features would you want to keep? We should discuss that first

**Comment by callumjt** [comment]
> > what if i were to merge this with "Reverse order of project controls" and change the name?
> 
> Merge "this"? What features would you want to keep? We should discuss that first

background colours, remove hover and the button width

**Comment by WorldLanguages** [comment]
I think it would be okay to rename "Reverse order of project controls" to "Customizable project controls" (or similar) and add three color selectors. But I don't know what others would think.

**Comment by WorldLanguages** [comment]
@callumjt We can see how to handle the credits later. Write it down as a TODO somewhere

**Comment by callumjt** [comment]
i just updated it

-merged with reverse order of project controls
-removed border radius
-removed button gap

**Comment by Secret-chest** [comment]
> > > Are they SVG? If then you can change the `fill` property
> > 
> > That won't work because they're `img` elements.
> 
> Oh, so should we reimplement them using inline SVG?

Any updates on this? Also why wouldn't you allow the size setting to go down to 24px? Some may want smaller buttons

**Comment by callumjt** [comment]
> 

1. i dont really know how to reimplement them as inline svg
2. sure i can make it 24 if you want

**Comment by BroJac5246** [comment]
Do we use American or British English? I'd assume the former, because Scratch is American. If that's the case, "Customisable" needs to be "Customizable," etc.

**Comment by callumjt** [comment]
> Do we use American or British English? I'd assume the former, because Scratch is American. If that's the case, "Customisable" needs to be "Customizable," etc.

Ah ok thank you

**Comment by WorldLanguages** [comment]
@callumjt Did you close this for any particular reason?

**Comment by callumjt** [comment]
> @callumjt Did you close this for any particular reason?

I remade my fork. I can open another one if you would like

**Comment by WorldLanguages** [comment]
@callumjt Oh, you don't need to remake your fork if you use branches.

**Comment by callumjt** [comment]
> @callumjt Oh, you don't need to remake your fork if you use branches. 

Ye but in the original one I made this in the main branch

**Comment by WorldLanguages** [comment]
I guess we should create an issue that suggests a feature to change the color of the green flag, pause buttons, and stop button icons.

**Pull request closed without merging**